File: day2.py
class Day2:

    def __init__(self):
        pass

    def final_state_left(self, content, v=12, n=2):
        pos = 0
        instructions = content[:]
        instructions[1] = v
        instructions[2] = n 
        while True:
            if instructions[pos] == 1:
                instructions[instructions[pos+3]] = instructions[instructions[pos+1]] + instructions[instructions[pos+2]]
            elif instructions[pos] == 2:
                instructions[instructions[pos+3]] = instructions[instructions[pos+1]] * instructions[instructions[pos+2]]
            elif instructions[pos] == 99:
                break
            pos += 4
        return instructions[0]

    def terminate_noun_verb(self, instructions):
        for i in range(100):
            for j in range(100):
                if self.final_state_left(instructions, v=i, n=j) == 19690720:
                    return i*100 + j 

if __name__ == "__main__":

    with open("input.txt",'r') as f:
        content = f.read()

    content = [int(i) for i in content.split(',')]

    print(Day2().final_state_left(content))
    print(Day2().terminate_noun_verb(content))
    

File: day2.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

class Day2{
    private:
        vector<int> inputs;

    public:
        Day2(string input_name){
            ifstream infile(input_name);
            string line;
            while (getline(infile, line, ',')){
                inputs.push_back(stoi(line));
            };
        };

        int final_state_left(int verb, int noun){
            vector<int> input;
            input = inputs;
            int pos = 0;
            input[1] = verb;
            input[2] = noun;
            while (true){
                if (input[pos] == 1){
                    input[input[pos+3]] = input[input[pos+1]] + input[input[pos+2]];
                }
                else if (input[pos] == 2) {
                    input[input[pos+3]] = input[input[pos+1]] * input[input[pos+2]];
                }
                else if (input[pos] == 99){
                    break;
                }
                pos += 4;
            }
            return input[0];
        }

        int terminate_noun_verb(){
            for (int i=0; i<100; i++){
                for (int j=0; j<100; j++){
                    if (final_state_left(i, j) == 19690720){
                        return 100 * i + j;
                    }
                }
            }

        }
};

int main(){
    
    Day2 day2 = Day2("input.txt");

    cout<< day2.final_state_left(12, 2) << endl;
    cout<< day2.terminate_noun_verb() << endl;

    return 0;
}